components/raftstore/src/store/fsm/peer.rs

```diff
@@ -4416,6 +4419,16 @@ where
                 {
                     alive_cache_idx = p.matched;
                 }
+                if *last_heartbeat < down_peer_limit {
```

Comment:
it will be better to add some unit test for this logic. It can be done later though.